api/kubernetes/v1alpha1/http_adapter_types.go

```diff
@@ -112,25 +112,27 @@ func (a *HTTPAdapter) Validate(data *api.Data) api.Problems {
 	return problems
 }
 
-func (a *HTTPAdapter) Resolve(data *api.Data) error {
+func (a *HTTPAdapter) Resolve(data *api.Data) (any, error) {
 	tpl := a.getTemplate()
 
 	url, err := tpl.URL.Resolve(data, true)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	headers := make(map[string]string, len(tpl.Headers))
 	for k, v := range tpl.Headers {
 		if headers[k], err = v.Resolve(data, true); err != nil {
-			return err
+			return nil, err
 		}
 	}
 
 	a.Spec.URL = url
 	a.Spec.Headers = headers
 
-	return nil
+	copy := a.DeepCopy().Spec
+
+	return &copy, nil
```

Comment:
The copy should be made first and then the output of the templates set on the copy. And probably only need to copy the spec.

```go
	copy := a.Spec.DeepCopy()

	copy.URL = url
	copy.Headers = headers

	return copy, nil
```

Reply:
Updated in https://github.com/xigxog/kubefox/pull/29/commits/cc9526c3a0e1fcfa278d5001efaa9dee2857fa72